core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableLayout.java

```diff
@@ -24,6 +24,7 @@ public class ConnectorTableLayout
     private final List<String> partitionColumns;
     private final boolean multipleWritersPerPartitionSupported;
 
+    @Deprecated
```

Comment:
I wouldn't say it's deprecated because `ConnectorTableLayout` is used for both read and writer path

Reply:
i couldn't find usages of this deprecated constructor on the read path.
but good point, i should update the usages